### ISSUE TITLE
getCurrentPosition is not strictly monotonic while playing

### ISSUE THREAD

**Issue by renaudcerrato** [androidx/media#1578]
### Version

Media3 1.4.0

### Devices that reproduce the issue

Pixel 7 Pro (Android 14)

### Devices that do not reproduce the issue

I assume none of them.

### Reproducible in the demo app?

Not tested

### Reproduction steps

I'm playing an MP4 audio while polling `getCurrentPosition` periodically to update the position, but I'm seeing some unexpected backward jump at around 3 seconds - the audio is playing fine - only the current position value seems wrong causing some unexpected UI glitches. It only seems to happen once, 3 seconds after playing started.

Confirmed using some debug traces:

```log
position: 2.901s
position: 2.917s
position: 2.934s
position: 2.951s
position: 2.853s <--- here
position: 2.87s
position: 2.887s
position: 2.903s
position: 2.92s
position: 2.938s
position: 2.956s
position: 2.975s
```

### Expected result

`getCurrentPosition` should be strictly monotonic during playing.

### Actual result

`getCurrentPosition` randomly jumps backward.

### Media

https://github.com/user-attachments/assets/65cf585e-a5b2-4920-966d-ac6b45a403fa

### Bug Report

- [ ] You will email the zip file produced by `adb bugreport` to android-media-github@google.com after filing this issue.

**Comment by tianyif** [comment]
@renaudcerrato, 

Thanks for filing the issue! Can you please share the audio with which you saw this problem? You can send it to android-media-github@google.com if you don't want to share it publicly.

**Comment by renaudcerrato** [comment]
Here it is:

https://github.com/user-attachments/assets/4835f933-6c63-4b01-aeca-929f517ad3a7

**Comment by tianyif** [comment]
Hi @renaudcerrato,

Thanks for sending the media. I quickly played it on the demo app and added some simple code to poll the `player.getCurrentPosition()` every 10 milliseconds (tried offload mode both on and off), but couldn't reproduce this issue. Did you see this problem every time when you play or from time to time? Did you turn on some special features on the player that doesn't come as default?

**Comment by renaudcerrato** [comment]
Seeing this everytime, on every audio files around 3 seconds after play(). If I pause() or seek(), same behavior 3 seconds after play().

BTW: my exoplayer is running into a playback service, the UI bind using a MediaSession. 

Will try to produce a minimal example as soon as possible.

**Comment by tianyif** [comment]
@renaudcerrato, 

> BTW: my exoplayer is running into a playback service, the UI bind using a MediaSession.

Thanks, this is really helpful information! Was this problem seen at your controller app? If yes, are you using media3 `MediaController`?

**Comment by renaudcerrato** [comment]
Yes, seeing it from MediaController

**Comment by tianyif** [comment]
Hi @renaudcerrato,

Unfortunately I still couldn't reproduce this issue on the demo session app (with adding some code to have a `controller` poll the `currentPosition` every 10ms), is it possible for you to provide a minimum sample project that can reproduce this issue?

**Comment by renaudcerrato** [comment]
Didn't had the time yet to produce a minimal POC due to holidays, but I'd be happy to privately share the project if you agree. Any means?

**Comment by tianyif** [comment]
@renaudcerrato, yes you can share it. But please provide enough information on how to navigate to the part of code related to this issue.

**Comment by renaudcerrato** [comment]
Just stripped down the project, here's a minimal working version (including recording to generate your sample): 
[recorder.zip](https://github.com/user-attachments/files/16622196/recorder.zip). Should be straightforward to navigate.

I've added some debug in the polling function in `src/recording/screen/player/MediaController.kt` :

![image](https://github.com/user-attachments/assets/88b51f1a-5981-4c9a-bdf8-29c1f8588fe6)

![image](https://github.com/user-attachments/assets/bc059cdb-64f5-48e2-96ed-a6cf4062be14)

This is a jetpack compose project, polling in a LaunchedEffect:

![image](https://github.com/user-attachments/assets/8ce8f052-3574-4c35-9910-4fca16b7307f)

**Comment by renaudcerrato** [comment]
I forgot to ping you @tianyif sorry

**Comment by andrewfarm** [comment]
I am seeing exactly the same behavior. In addition, even if the app is *not* calling `getCurrentPosition` itself, if the app uses a [`MediaSessionService`](https://developer.android.com/media/media3/session/background-playback) to play media in the background, the `MediaStyle` notification provided by the library exhibits the same behavior: the playback position slider jumps back slightly at the 3 second mark. (This can be difficult to notice because the notification's playback position slider only updates several times per second.)

**Comment by nift4** [comment]
I am also seeing this behaviour with `MediaController` so I believe this is may be related to media3-session library, any chance someone could take a look?
